lld/COFF/Writer.cpp

```diff
@@ -2062,6 +2062,20 @@ void Writer::createECChunks() {
     if (auto thunk = dyn_cast<ECExportThunkChunk>(sym->getChunk())) {
       hexpthkSec->addChunk(thunk);
       exportThunks.push_back({thunk, thunk->target});
+    } else if (auto def = dyn_cast<DefinedRegular>(sym)) {
+      // Allow section chunk to be treated as an export thunk if it looks like
+      // one.
+      SectionChunk *chunk = def->getChunk();
+      if (!chunk->live || chunk->getMachine() != AMD64)
+        continue;
+      assert(sym->getName().starts_with("EXP+"));
+      StringRef targetName = sym->getName().substr(strlen("EXP+"));
+      Symbol *targetSym = ctx.symtab.find((targetName + "$hp_target").str());
+      if (!targetSym)
+        targetSym = ctx.symtab.find(targetName);
```

Comment:
Maybe add a comment here, that for an export thunk `EXP+#foo`, we're first looking for `#foo$hp_target`, but secondarily also for `#foo`. (I was wondering about this and scratching my head, overlooking the fact that we do look for the unsuffixed version two lines below...)

Reply:
Good point, I added a comment and merged. Thanks for reviews!